File: backend/src/models/RecommendationHistory.ts
import mongoose, { Schema, Document } from 'mongoose';
import { BaseModelPlugin } from './BaseModelPlugin';
import crypto from 'crypto';
import { IOptimizationRecommendation } from './OptimizationRecommendation';

export interface IRecommendationHistory extends Document<any> {
    _id: Buffer | string;
    recommendation: IOptimizationRecommendation['_id'] | any;
    action_taken: string;
    accepted: boolean;
    implemented_at: Date | null;
    verified_at: Date | null;
}

const recommendationHistorySchema = new Schema<IRecommendationHistory>({
    _id: { type: 'UUID', default: () => crypto.randomUUID() },
    recommendation: { type: 'UUID', ref: 'OptimizationRecommendation', required: true },
    action_taken: { type: String, required: true },
    accepted: { type: Boolean, required: true, index: true },
    implemented_at: { type: Date, default: null },
    verified_at: { type: Date, default: null }
}, { collection: 'recommendation_history' });

recommendationHistorySchema.plugin(BaseModelPlugin);

export const RecommendationHistory = mongoose.model<IRecommendationHistory>('RecommendationHistory', recommendationHistorySchema);

File: backend/app/models/recommendation_history.py
from datetime import datetime
from typing import Optional
from pydantic import Field
from beanie import Indexed, Link
from app.models.base import BaseUUIDDocument
from app.models.optimization_recommendation import OptimizationRecommendation
from app.models.base import get_utc_now

class RecommendationHistory(BaseUUIDDocument):
    recommendation: Link[OptimizationRecommendation]
    action_taken: str
    accepted: bool
    implemented_at: Optional[datetime] = None
    verified_at: Optional[datetime] = None

    class Settings:
        name = "recommendation_history"
        indexes = [
            "accepted"
        ]
